src/foam/nanos/theme/Theme.js

```diff
@@ -754,6 +763,23 @@ foam.CLASS({
           value1.copyFrom(value2);
         }
       `
+    },
+    {
+      name: 'isCapabilityRestricted',
+      type: 'Boolean',
+      args: [ 'String capId' ],
+      code: function(capId) {
+        if ( this.restrictedCapabilities != null ) {
+          return this.restrictedCapabilities.includes(capId)
+        }
+        return false;
+      },
+      javaCode: `
+        if ( getRestrictedCapabilities() != null ) {
+          return (new HashSet<>(Arrays.asList(getRestrictedCapabilities()))).contains(capId);
```

Comment:
This is still going to create a new HashSet() for each call and copy all of the data. There should be a transient variable like restrictedCapabilitiesSet_ whose factory creates the hashset so that it can be reused. The postSet on restrictedCapabilities should clear that property.